src/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncMethodBuilder.cs

```diff
@@ -721,16 +715,8 @@ internal static class AsyncMethodBuilderCore
         /// <exception cref="System.ArgumentNullException">The <paramref name="stateMachine"/> argument is null (Nothing in Visual Basic).</exception>
         [DebuggerStepThrough]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal static void Start<TStateMachine>(ref TStateMachine stateMachine)
-            where TStateMachine : IAsyncStateMachine
-        {
-            // Async state machines are required not to throw, so no need for try/finally here.
-            Thread currentThread = Thread.CurrentThread;
-            ExecutionContextSwitcher ecs = default(ExecutionContextSwitcher);
-            ExecutionContext.EstablishCopyOnWriteScope(currentThread, ref ecs);
-            stateMachine.MoveNext();
-            ecs.Undo(currentThread);
-        }
+        internal static void Start<TStateMachine>(ref TStateMachine stateMachine) where TStateMachine : IAsyncStateMachine
+            => AsyncMethodBuilder.Start(ref stateMachine);
```

Comment:
The AsyncMethodBuilder vs. AsyncMethodBuilderCore split does not make much sense. Also, this code is filled with landmines like debugger depending on names of these internal types (https://github.com/dotnet/coreclr/blob/181680249af92aa45c8fb44a3ee865c7b61fd765/src/mscorlib/src/System/Runtime/CompilerServices/AsyncMethodBuilder.cs#L895).

Could you please move the implementation of `Start<TStateMachine>` to here, and not have AsyncMethodBuilder at all?

Reply:
Changed